types/index.d.ts

```diff
@@ -0,0 +1,90 @@
+import Vue, { PluginFunction, VueConstructor } from "vue"
+
+declare const VueGtm: VueGtm
+export default VueGtm
+export interface VueGtm {
+  install: PluginFunction<VueGtmUseOptions>
+}
```

Comment:
This library exports a plugin object, not a plugin function. As such:

```suggestion
import Vue, { PluginObject, VueConstructor } from "vue"

declare const VueGtm: VueGtmPlugin
export default VueGtm
export interface VueGtmPlugin extends PluginObject<VueGtmUseOptions> {
}
```

Reply:
https://github.com/mib200/vue-gtm/blob/6e5d5e3ff49e2ddd75164f98c7134291c769ac1a/src/index.js#L11

Please explain why you think that it exports a `PluginObject`?

Reply:
Because it does: 
![image](https://user-images.githubusercontent.com/5017521/77666179-d9717100-6f3d-11ea-8e60-76c363e0da78.png)

This line is not directly exporting the install function - it is exporting an object with one member, `install`, whose value is the `install` function defined above.
https://github.com/mib200/vue-gtm/blob/6e5d5e3ff49e2ddd75164f98c7134291c769ac1a/src/index.js#L77

`VueGtm` in the old diff already more or less a `PluginObject` - you're just not utilizing the interface that Vue already provides.

Additionally, I renamed the interface for the object because Typescript can have issues in some cases where you try to export both a type `interface VueGtm` and a value `const VueGtm` with the same name.